### ISSUE TITLE
Error rather than fallback when a V3 or later endpoint is not present in exchange capabilities

### ISSUE THREAD

**Issue by realbigsean** [sigp/lighthouse#4766]
## Description

We check `engine_exchangeCapabilities` for the latest availabile version and fallback to prior versions based on what is available. The behavior made sense when endpoints were backwards compatible, but V3 and later endpoints are no longer backwards compatible, so querying older versions when a method doesn't apper no longer makes sense. I think we should fail loudly instead of attempting a likely fruitless (and unexpected) fallback.

**Comment by jimmygchen** [comment]
I think we're already doing this? Or perhaps I'm missing something?

`newPayload`:
https://github.com/sigp/lighthouse/blob/7d468cb487dfbd753f8079910efc7a53a00da86a/beacon_node/execution_layer/src/engine_api/http.rs#L1098-L1104

`getPayload`:
https://github.com/sigp/lighthouse/blob/7d468cb487dfbd753f8079910efc7a53a00da86a/beacon_node/execution_layer/src/engine_api/http.rs#L1126-L1132

`forkchoiceUpdated`:
https://github.com/sigp/lighthouse/blob/7d468cb487dfbd753f8079910efc7a53a00da86a/beacon_node/execution_layer/src/engine_api/http.rs#L1161-L1170

**Comment by realbigsean** [comment]
Good point, I hadn't looked deeply into this yet. Looks like only time this happens is when there's no payload attributes on an FCU:
```
        if let Some(payload_attributes) = maybe_payload_attributes.as_ref() {
...
        } else if engine_capabilities.forkchoice_updated_v3 {
            self.forkchoice_updated_v3(forkchoice_state, maybe_payload_attributes)
                .await
        } else if engine_capabilities.forkchoice_updated_v2 {
            self.forkchoice_updated_v2(forkchoice_state, maybe_payload_attributes)
```

**Comment by realbigsean** [comment]
stale
